Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp

```diff
@@ -136,51 +137,76 @@ namespace AZ
+                    materialTypeJobDependency.m_sourceFile.m_sourceFileDependencyPath = resolvedMaterialTypePath;
+                    materialTypeJobDependency.m_jobKey = MaterialTypeBuilder::FinalStageJobKey;
+                    materialTypeJobDependency.m_productSubIds = { 0 };
+                    MaterialBuilderUtils::AddFingerprintForDependency(
+                        materialTypeJobDependency.m_sourceFile.m_sourceFileDependencyPath, outputJobDescriptor);
+
+                    for (const auto& shader : materialTypeSourceData.GetShaderReferences())
+                    {
+                        auto& shaderJobDependency = outputJobDescriptor.m_jobDependencyList.emplace_back();
+                        shaderJobDependency.m_type = AssetBuilderSDK::JobDependencyType::Order;
+                        shaderJobDependency.m_sourceFile.m_sourceFileDependencyPath =
+                            AssetUtils::ResolvePathReference(resolvedMaterialTypePath, shader.m_shaderFilePath);
+                        shaderJobDependency.m_jobKey = "Shader Asset";
+                        MaterialBuilderUtils::AddFingerprintForDependency(
+                            shaderJobDependency.m_sourceFile.m_sourceFileDependencyPath, outputJobDescriptor);
+                    }
                 }
                 else if (materialTypeFormat == MaterialTypeSourceData::Format::Abstract)
                 {
-                    MaterialBuilderUtils::AddPossibleDependencies(
-                        materialSourcePath,
-                        resolvedMaterialTypePath,
-                        response,
-                        outputJobDescriptor,
-                        MaterialTypeBuilder::PipelineStageJobKey,
-                        AssetBuilderSDK::JobDependencyType::Order,
-                        {},
-                        AssetBuilderSDK::CommonPlatformName);
-
-                    const AZStd::string intermediateMaterialTypePath =
-                        MaterialUtils::PredictIntermediateMaterialTypeSourcePath(materialSourcePath, resolvedMaterialTypePath);
+                    // Create a dependency on the abstract, pipeline, version of the material type and its products. The pipeline based
+                    // material type builder uses the 'common' asset platform ID because it produces immediate assets. The sub ID filter
+                    // should remain empty to observe all produced intermediate assets.
+                    auto& materialTypeJobDependency = outputJobDescriptor.m_jobDependencyList.emplace_back();
+                    materialTypeJobDependency.m_type = AssetBuilderSDK::JobDependencyType::Order;
+                    materialTypeJobDependency.m_sourceFile.m_sourceFileDependencyPath = resolvedMaterialTypePath;
+                    materialTypeJobDependency.m_jobKey = MaterialTypeBuilder::PipelineStageJobKey;
+                    materialTypeJobDependency.m_platformIdentifier = AssetBuilderSDK::CommonPlatformName;
```

Comment:
my eyes gloss over when looking at all these blocks of code.  I wish we had a simpler way to express a dependency e.g.
````
auto materialTypeJobDependency = OrderDependency( resolvedMaterialTypePath, PipelinStageJobKey );
```

or
```
auto materialTypeJobDependency {
    resolvedMaterialTypePath,
    PipelinStageJobKey,
    CommonPlatformName,
    JobDependencyType::Order
};
```

Reply:
That is part of what the previous function utility function was supposed to do but it did a lot of extra guesswork trying to compensate for missing or yet to be created intermediate assets. I could make a function that does just this for all the cases that follow the exact pattern.

Reply:
I'm fine with that change being in another PR, I'd rather get the fix in, then delay it